nomt/src/beatree/allocator/free_list.rs

```diff
@@ -187,15 +187,16 @@ impl FreeList {
         // append the released free list pages
         to_push.extend(self.released_portions.drain(..));
 
-        if had_pops && to_push.is_empty() {
+        let pages = if had_pops && to_push.is_empty() {
             // note: empty vec when head is empty.
-            return self.encode_head(page_pool).into_iter().collect();
-        }
-
-        let new_pages = self.preallocate(&mut to_push, bump);
-        let pages = self.push_and_encode(page_pool, &to_push, new_pages);
-        // preallocate pops, therefore, we must set it back.
-        self.pop = false;
+            self.encode_head(page_pool).into_iter().collect()
```

Comment:
I don't fully understand this.

If `to_push` is empty and `had_pops` is false, then we are dealing with an empty commit. Then, shouldn't the head of the free-list be exactly equal to what it was before (i.e. the last written out head)?